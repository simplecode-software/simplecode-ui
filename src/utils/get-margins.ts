import React from 'react';
import createCachedSelector from 're-reselect';

import { getCssSides } from './get-css-sides';


export const getMargins = createCachedSelector<
    string | number | null,
    string | number | null,
    React.CSSProperties | null
>(
    (value) => value,
    (value) => getCssSides('margin', value),
)((value) => `getMargins-${value}`);
