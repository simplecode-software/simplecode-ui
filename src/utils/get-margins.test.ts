import { getMargins } from './get-margins';


describe('Get Margins', () => {
    it('is a function', () => {
        expect(typeof getMargins).toBe('function');
    });
    
    it('computes single margins', () => {
        expect(getMargins('10')).toMatchSnapshot();
    });
    
    it('computes double margins', () => {
        expect(getMargins('10 20')).toMatchSnapshot();
    });
    
    it('computes tripple margins', () => {
        expect(getMargins('10 20 30')).toMatchSnapshot();
    });
    
    it('computes full margins', () => {
        expect(getMargins('10 20 30 40')).toMatchSnapshot();
    });
    
    it('trims extra margins', () => {
        expect(getMargins('10 20 30 40 50')).toMatchSnapshot();
    });
    
    it('computes number margin', () => {
        expect(getMargins(40)).toMatchSnapshot();
    });
    
    it('returns null if no margin', () => {
        expect(getMargins(null)).toMatchSnapshot();
    });
    
    it('returns null if empty string', () => {
        expect(getMargins('')).toMatchSnapshot();
    });
});
