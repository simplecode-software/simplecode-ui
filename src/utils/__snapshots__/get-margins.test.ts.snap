// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Margins computes double margins 1`] = `
{
  "marginBottom": 10,
  "marginLeft": 20,
  "marginRight": 20,
  "marginTop": 10,
}
`;

exports[`Get Margins computes full margins 1`] = `
{
  "marginBottom": 30,
  "marginLeft": 40,
  "marginRight": 20,
  "marginTop": 10,
}
`;

exports[`Get Margins computes number margin 1`] = `
{
  "margin": 40,
}
`;

exports[`Get Margins computes single margins 1`] = `
{
  "marginBottom": 10,
  "marginLeft": 10,
  "marginRight": 10,
  "marginTop": 10,
}
`;

exports[`Get Margins computes tripple margins 1`] = `
{
  "marginBottom": 30,
  "marginLeft": 20,
  "marginRight": 20,
  "marginTop": 10,
}
`;

exports[`Get Margins returns null if empty string 1`] = `null`;

exports[`Get Margins returns null if no margin 1`] = `null`;

exports[`Get Margins trims extra margins 1`] = `
{
  "marginBottom": 30,
  "marginLeft": 40,
  "marginRight": 20,
  "marginTop": 10,
}
`;
